test/unit/components/uploadlaunchoverlay.test.js

```diff
@@ -0,0 +1,91 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ */
+/* global chai */
+/* global describe */
+/* global sinon */
+/* global it */
+/* global beforeEach */
+
+import React from 'react';
+import { mount } from 'enzyme';
+
+import UploadLaunchOverlay from '../../../app/components/uploadlaunchoverlay';
+import ModalOverlay from '../../../app/components/modaloverlay';
+import { URL_UPLOADER_DOWNLOAD_PAGE } from '../../../app/core/constants';
+
+const expect = chai.expect;
+
+import releases from '../../fixtures/githubreleasefixture';
+
+describe('UploadLaunchOverlay', function () {
+  const props = {
+    overlayClickHandler: sinon.spy()
+  };
+
+  let wrapper;
+  beforeEach(() => {
+    wrapper = mount( <UploadLaunchOverlay {...props} /> );
+  });
+
+  it('should be a function', function() {
+    expect(UploadLaunchOverlay).to.be.a('function');
+  });
+
+  describe('getLatestGithubRelease', function() {
+    it('should return the latest github release from a list of releases', function() {
+      expect(wrapper.instance().getLatestGithubRelease(releases)).to.deep.equal({
+        latestWinRelease: 'https://github.com/tidepool-org/chrome-uploader/releases/download/v0.309.0/tidepool-uploader-setup-0.309.0.exe',
+        latestMacRelease: 'https://github.com/tidepool-org/chrome-uploader/releases/download/v0.309.0/tidepool-uploader-0.309.0.pkg'
+      });
+    });
+  });
+
+  describe('render', function() {
+    it('should render without problems', function () {
+      expect(wrapper.find(UploadLaunchOverlay)).to.have.length(1);
+      expect(wrapper.find(ModalOverlay)).to.have.length(1);
+    });
+
+    it('should respond to an onClick event', () => {
+      var callCount = props.overlayClickHandler.callCount;
+      wrapper.find('.ModalOverlay-target').simulate('click');
+      expect(props.overlayClickHandler.callCount).to.equal(callCount + 1);
+    });
+
+    it('should have disabled download buttons if no URLs have been set', () => {
+      wrapper.setState({
+        latestWinRelease: null,
+        latestMacRelease: null
+      });
+      expect(wrapper.find('a.disabled')).to.have.length(2);
+    });
+
+    it('should have active buttons if URLs have been set', () => {
+      wrapper.setState({
+        latestMacRelease: 'test',
+        latestWinRelease: 'test'
+      });
+      expect(wrapper.find('a')).to.have.length(2);
+      expect(wrapper.find('a.disabled')).to.have.length(0);
+    });
+
+    it('should display download link if error retrieving github releases', () => {
+      wrapper.setState({
+        error: 'some error'
+      });
+      expect(wrapper.find(`a[href=${URL_UPLOADER_DOWNLOAD_PAGE}]`)).to.have.length(0);
```

Comment:
Should this not be expecting the length to be 1?  Just setting it to 1 would fail, though, as there's an issue with enzyme and href matching, so you'll need to change it. see https://github.com/airbnb/enzyme/issues/283#issuecomment-278086218

Changing to this, for instance, passes:
`expect(wrapper.find({ href: URL_UPLOADER_DOWNLOAD_PAGE }).filter('a')).to.have.length(1);`